alacritty_terminal/src/ansi.rs

```diff
@@ -25,62 +25,49 @@ use vte;
 use crate::term::color::Rgb;
 
 // Parse color arguments
-//
-// Expect that color argument looks like "rgb:xx/xx/xx" or "#xxxxxx"
-fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
-    let mut iter = color.iter();
 
-    macro_rules! next {
-        () => {
-            iter.next().map(|v| *v as char)
-        };
+fn is_rgb_separator(ch: Option<char>) -> bool {
+    ch == None || ch == Some('/') || ch == Some('\x07')
+}
+
+// Expect that color argument looks like "rgb:x[xxx]/x[xxx]/x[xxx]" or "#xxxxxx"
+fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
+    let len = color.len();
+    // Legacy colors are in the form "#xxxxxx"
+    // The use of this format is no longer encouraged by xparsecolor
+    let mut is_legacy = false;
+
+    if len >= 2 && color[0] == b'#' {
```

Comment:
This should probably be `>= 4`, since `#RGB` is at least 4 characters.

Reply:
[Updated](https://github.com/rbong/alacritty/blob/cdbcd8357fc2955c011785ed02bd93b386509561/alacritty_terminal/src/ansi.rs#L39-L43). I used 5 instead and started using the bell character instead, and modified the length check for non-legacy colors as well. Tests have been updated to use the bell character.